packages/core/src/sync-historical/index.ts

```diff
@@ -133,6 +152,91 @@ export const createHistoricalSync = async (
   // Helper functions for specific sync tasks
   ////////
 
+  /**
+   * Split "eth_getLogs" requests into ranges inferred from errors.
+   */
+  const logRequestHelper = async ({
+    address,
+    topics,
+    interval,
+    filter,
+  }: {
+    address?: Address | Address[];
+    topics?: LogTopic[];
+    interval: Interval;
+    filter: LogFilter | LogFactory;
```

Comment:
Aren't the address and topics available on the `filter` ?

Reply:
Yes, going to update this a bit, but it's not as straight-forward as just using `filter` because of factory contracts